pkg/cmd/roachtest/roachtestutil/disconnect.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package roachtestutil
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/stretchr/testify/require"
+)
+
+// Disconnect takes a set of nodes and each nodes internal ips. It disconnects
+// each node from all the others in the list.
+func Disconnect(t test.Test, c cluster.Cluster, ctx context.Context, nodes []int) {
```

Comment:
❓ I wonder if `Disconnect(t test.Test, c cluster.Cluster, ctx context.Context, node int, toDisconnect []int)` where we disconnect `node` from all nodes in `toDisconnect` would be a clearer API overall.  Adding iptable rules on both sides shouldn't be necessary.   Given the usage in this PR, let's not change it now, just thinking "out loud"